File: nodejs/src/Level1.js
const Level = require('./Level.js')
class Level1 extends Level
{
    constructor()
    {
        super(1)
    }

    run()
    {
        let rentals = this.input.rentals.map(rental => {
            let duration = this.getDuration(rental)
            let car = this.cars[rental.car_id]
            
            return {
                id: rental.id,
                price: duration * car.price_per_day + rental.distance * car.price_per_km
            }
        })

        return {rentals: rentals}
    }
}

module.exports = Level1

File: elixir/lib/level1.ex
defmodule Drivy.Level1 do
  import Drivy

  def run(input) do
    cars = to_map(input["cars"])
    rentals = input["rentals"]

    rentals =
      Enum.map(rentals, fn r ->
        car = Map.get(cars, r["car_id"])
        %{"id" => r["id"], "price" => get_price(car, r)}
      end)

    %{"rentals" => rentals}
  end

  defp get_price(car, rental) do
    car["price_per_km"] * rental["distance"] +
      car["price_per_day"] * to_days(rental["start_date"], rental["end_date"])
  end
end
